docs/src/main/tut/deployment.md

```diff
@@ -0,0 +1,148 @@
+
+To create a truly static native image we need to use muslC according to [issue](https://github.com/oracle/graal/issues/1919#issuecomment-589085506) . Instructions and an example bundle are provided [here](https://github.com/gradinac/musl-bundle-example). For the sake of our example we can download the resulting bundle for our build. We will need to use the path to the unpacked bundle as an argument to build the image.
+
+```
+> wget https://github.com/gradinac/musl-bundle-example/releases/download/v1.0/musl.tar.gz
+> tar -xf musl.tar.gz
+```
+
+### META-INF resources for reflection
+
+We need META-INF resources where we use reflection. In general reflection isn't used throughout http4s, however it is used by some of the logging dependencies. We've provided a native image example in the [http4s gitter8 template](https://github.com/http4s/http4s.g8)
+
+#### Build an assembled jar using GRAALVM
+
+After installing the above dependencies you should build an assembled JAR. We can again use `sbt assembly` or your favorite build tool / plugin to create the assembled jar. The important thing is that we should be using the GraalVM version of JAVA to do so.
+
+#### Create the native image with the assembled JAR
+
+After we have built the assembled JAR containing all our JAVA dependencies, we use that JAR to build our native image. In the command below we need to replace the muslC and assembly jar paths with the appropriate locations.
+
+```
+native-image --static -H:+ReportExceptionStackTraces -H:UseMuslC="/path.to/muslC" --allow-incomplete-classpath --no-fallback --initialize-at-build-time --enable-http --enable-https --enable-all-security-services --verbose -jar "./path.to.assembly.jar" projectBinaryImage
+```
+
+A breakout for the command parameters (image generation options) :
+
+`-H:UseMuslC` to fix DNS and related segfaults and build a true static image that doesn't depend on linked libraries. 
+
+`--initialize-at-build-time` again is related to building the static image. Again we lose performance if we instead do this at runtime.
+
+`--enable-http`, `--enable-https` should correspond to related protocols
+
+`--enable-all-security-services` will likely be required to make or receive https requests.
+
+Additional image generation options are found in the [native image reference](https://www.graalvm.org/docs/reference-manual/native-image/)
+
+
+#### Execute the native image
+
+Finally we should be able to execute our output `projectBinaryImage`
+
+```
+./projectBinaryImage
+```
+
+At this point we may want to package the binary in a `docker` container, integrate the image generation to a greater build process, create `init` scripts via `systemd`, etc.
+
+#### Why static? 
+
+As an alternative to executing via the `JVM` `GraalVM native-image` allows us to execute as a native binary. For example in Linux environments this might be known as an `ELF` format. There are a number of ways to generate a native image be it `dynamic` or `static`. 
```

Comment:
```suggestion
As an alternative to executing via the JVM, GraalVM's native-image allows us to execute as a native binary. For example, in Linux environments this might be known as an `ELF` format. There are a number of ways to generate a native image, be it dynamic or static. 
```